modules/udparsers/m2c00/spcn_trace.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env python3
+
+"""
+This module contains functions to parse and format SPCN Trace data produced by
+IO drawers.
+
+The SPCN Trace data can be obtained in the following ways:
+    * Using the web interface to the BMC and predecessors of the BMC.
+    * Collecting a Resource Dump.  The SPCN Trace data must be extracted from
+      the dump using a separate Resource Dump tool.
+
+The SPCN Trace is a hex dump of memory containing the following:
+    * ILOG/PTE data
+    * Zero or more trace buffers
+
+This module can also be run standalone as a script.
+"""
+
+import argparse
+import sys
+
+import pel.hexdump as hexdump
+from udparsers.m2c00.ilog import parse_ilog_data
+from udparsers.m2c00.trace import parse_trace_data
+
+
+# Byte values expected at the start of a trace buffer header
+TRACE_BUFFER_HEADER_START = (b'\x02'        # ver field
```

Comment:
is this data also in the trace module?  Could you get the constant from there?

Reply:
It's not currently defined in the trace module.  I tried to make that trace module fairly generic, so it could in theory support generic FSP trace buffers too.

The constants here are a bit of a hack, such as a hardcoded version field value.  The hack is necessary because the ILOG data doesn't have a header that defines the data length.

I could move the constants to the trace module if you think that would make more sense.

Reply:
I moved the trace buffer names into the trace module.  I left the hardcoded buffer header start bytes, as that is a hack required only for the SPCN Trace functionality.